esmvaltool/utils/recipe_filler.py

```diff
@@ -183,13 +183,13 @@ def _normalize_path(path):
     return os.path.abspath(os.path.expanduser(os.path.expandvars(path)))
 
 
-def read_config_user_file(config_file, folder_name, options=None):
-    """Read config user file and store settings in a dictionary."""
+def read_config_file(config_file, folder_name, options=None):
```

Comment:
It looks like this whole function could be replaced by `from esmvalcore.config import CFG`, could you have a look?

Reply:
Good point! I think this is possible, but when trying to do that, I encountered the following problem: there's lots of custom code here to read the config-developer file and to find files, e.g., [here](https://github.com/ESMValGroup/ESMValTool/blob/7b1cd473e62326ca26e084c16d57da79644a6d11/esmvaltool/utils/recipe_filler.py#L301-L336). This could probably all be replaced with appropriate methods from `esmvalcore` here.

This made me wondering: do we actually still need this functionality? I think every feature in `recipe_filler.py` can be replaced with a proper wildcard syntax, e.g., [here](https://github.com/ESMValGroup/ESMValCore/blob/main/notebooks/discovering-data.ipynb). I guess the wildcard syntax also solves some of the limitations of the recipe filler (e.g., from the docs: "the tool doesn’t yet work with derived variables; it will not return any available datasets", "operation restricted to CMIP data only, OBS lookup is not available yet.")

@valeriupredoi @bouweandela do you agree? Are you aware of anyone that uses this regularly?

Reply:
I agree, I think recipe_filler can be retired. Maybe we should check with @ledm?

Reply:
PR: https://github.com/ESMValGroup/ESMValTool/pull/3777

(in case we agree to remove it)

Reply:
How about I remove all changes from the recipe filler code in this PR so we can proceed with this? Would be great to finally fix our doc build again. If we decide to keep the recipe filler in https://github.com/ESMValGroup/ESMValTool/pull/3777, I will make the appropriate changes to it in a separate PR.